lib/jekyll-paginate-v2/generator/paginationModel.rb

```diff
@@ -227,12 +227,14 @@ def paginate(template, config, site_title, all_posts, all_tags, all_categories,
             puts "Pagination: ".rjust(20) + "Rolling through the date fields for all documents"
           end
           for p in using_posts
-            tmp_date = p.date
-            if( !tmp_date || tmp_date.nil? )
-              if @debug
-                puts "Pagination: ".rjust(20) + "Explicitly assigning date for doc: #{p.data['title']} | #{p.path}"
+            if p.respond_to?('date')
```

Comment:
Hey Peter, sorry for the long delay I have been a little busy with real life these days. I like this solution but you're right I will have to check how it will interact with the `--drafts`` bug this ugly codeblock is attempting to "fix"